newsletter/views.py

```diff
@@ -1,8 +1,23 @@
-from . import models, serializers
-from rest_framework.generics import CreateAPIView
+from rest_framework.generics import CreateAPIView, UpdateAPIView
+from django.http import HttpResponse
+from .models import NewsletterSubscriber
+from .serializers import NewsletterSubscriberSerializer
 
 
-# Create your views here.
+# View for creating a new newsletter subscription
 class NewsletterCreateView(CreateAPIView):
-    queryset = models.NewsletterSubscriber.objects.all()
-    serializer_class = serializers.NewsletterSubscriber
+    queryset = NewsletterSubscriber.objects.all()
+    serializer_class = NewsletterSubscriberSerializer
+
+
+# View for unsubscribing using the unique unsubscribe token
+class UnsubscribeView(UpdateAPIView):
+    queryset = NewsletterSubscriber.objects.all()
+    serializer_class = NewsletterSubscriberSerializer
+    lookup_field = 'unsubscribe_token'
+
+    def update(self, request, *args, **kwargs):
```

Comment:
Lets do this logic in the serializers @Sailesh-Singh 
@rex9840 will help on this one. 

Quite simple but lets not do business logic in `views.py` from now on.